pennylane/ops/op_math/dot.py

```diff
@@ -0,0 +1,60 @@
+# Copyright 2018-2023 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+This file contains the definition of the dot function, which computes the dot product between
+a vector and a list of operators.
+"""
+from typing import Sequence
+
+import pennylane as qml
+from pennylane.operation import Operator
+from pennylane.ops import Hamiltonian
+
+
+def dot(coeffs: Sequence[float], ops: Sequence[Operator], hamiltonian=True):
```

Comment:
Do we even want the option to return a Hamiltonian? If someone wants a Hamiltonian, they don't need this function.  This utility function would mainly be for operator arithmetic usage.

So in my opinion, I would either set `hamiltonian=False` by default or just always remove an operator arithmetic object.

Reply:
Removed `hamiltonian` kwarg.

Added `pauli` kwarg. When `True` it returns a `PauliSentence`.

Reply:
Check the PR description for more information.